client/layout/controller.js

```diff
@@ -53,14 +54,35 @@ class Controller extends Component {
 		const search = this.props.location.search.substring( 1 );
 		const query = parse( search );
 		const page = find( getPages(), { path } );
+		window.wpNavMenuUrlUpdate( page.wpMenu, query );
 		window.wpNavMenuClassChange( page.wpMenu, this.props.location.pathname );
 		return createElement( page.container, { params, path: url, pathMatch: path, query } );
 	}
 }
 
+// Update links in wp-admin menu to persist time related queries
+window.wpNavMenuUrlUpdate = function( menuClass, { period, compare, after, before } ) {
+	const timeRelatedQuery = omitBy( { period, compare, after, before }, isUndefined );
+	const search = stringifyQuery( timeRelatedQuery );
+
+	Array.from( document.querySelectorAll( `#${ menuClass } a` ) ).forEach( item => {
+		/**
+		 * Example href:
+		 *
+		 * http://example.com/wp-admin/admin.php?page=wc-admin#/analytics/orders?period=today&compare=previous_year
+		 */
+		if ( -1 !== item.href.indexOf( 'wc-admin#' ) ) {
```

Comment:
Given that we are using Babel, I usually prefer using the ES6 method `.includes()`:
```suggestion
		if ( item.href.includes( 'wc-admin#' ) ) {
```